pkg/apis/core/validation/shoot.go

```diff
@@ -168,7 +168,7 @@ func validateShootKubeconfigRotation(newMeta, oldMeta metav1.ObjectMeta, fldPath
 }
 
 // ValidateShootSpec validates the specification of a Shoot object.
-func ValidateShootSpec(meta metav1.ObjectMeta, spec *core.ShootSpec, fldPath *field.Path, inTemplate bool) field.ErrorList {
+func ValidateShootSpec(meta metav1.ObjectMeta, spec *core.ShootSpec, fldPath *field.Path, inTemplate, isUpdate bool) field.ErrorList {
```

Comment:
Why do you introduce an `isUpdate` parameter here when there is already a `ValidateShootSpecUpdate` function? This confusing. Can't you run your function in https://github.com/gardener/gardener/pull/5389/files#diff-a04dba2257c8893abf53b88a89b74aad4031f17b636f5d1e975e5b1dc8daa835R123?